osf/models/node.py

```diff
@@ -1801,14 +1803,15 @@ def register_node(self, schema, auth, data, parent=None):
         for node_relation in original.node_relations.filter(child__is_deleted=False):
             node_contained = node_relation.child
             # Register child nodes
-            if not node_relation.is_node_link:
+            if not node_relation.is_node_link and node_contained._id not in excluded_node_ids:
```

Comment:
Minor: Rather than checking `node_contained._id not in excluded_node_ids` twice, I think it'd be more future-proof to break out early:

```python
for node_relation in ...:
    if node_contained._id in excluded_node_ids:
        continue
...
```